mountainsort/src/processors/fit_stage.cpp

```diff
@@ -114,7 +113,8 @@ bool fit_stage(const QString& timeseries_path, const QString& firings_path, cons
             if (to_use[i] == 1) {
                 something_changed = true;
                 num_added++;
-                subtract_scaled_template(M * T, X.dataPtr(0, times_to_try[i] - Tmid), templates.dataPtr(0, 0, labels_to_try[i] - 1));
+                long tt = (long)(times_to_try[i] - Tmid + 0.5);
```

Comment:
You don't have to make an explicit cast here. Since lvalue is of type long, the cast will be done implicitly.